include/alpaka/pltf/PltfGenericSycl.hpp

```diff
@@ -0,0 +1,689 @@
+/* Copyright 2022 Jan Stephan
+ *
+ * This file is part of Alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+#ifdef ALPAKA_ACC_SYCL_ENABLED
+
+#    include <alpaka/core/Concepts.hpp>
+#    include <alpaka/core/Sycl.hpp>
+#    include <alpaka/dev/Traits.hpp>
+#    include <alpaka/pltf/Traits.hpp>
+
+#    include <CL/sycl.hpp>
+
+#    include <iostream>
+#    include <sstream>
+#    include <stdexcept>
+#    include <vector>
+
+namespace alpaka::experimental
+{
+    //! The SYCL device manager.
+    class PltfGenericSycl : public concepts::Implements<ConceptPltf, PltfGenericSycl>
+    {
+    public:
+        PltfGenericSycl() = delete;
+    };
+} // namespace alpaka::experimental
+
+namespace alpaka::traits
+{
+    //! The SYCL platform device count get trait specialization.
+    template<typename TPltf>
+    struct GetDevCount<TPltf, std::enable_if_t<std::is_base_of_v<experimental::PltfGenericSycl, TPltf>>>
+    {
+        static auto getDevCount() -> std::size_t
+        {
+            ALPAKA_DEBUG_FULL_LOG_SCOPE;
+
+            static auto dev_num = std::size_t{0};
+            if(dev_num == 0)
+            {
+                auto platform = sycl::platform{typename TPltf::selector{}};
+                auto devices = platform.get_devices();
+                dev_num = devices.size();
+            }
```

Comment:
```suggestion
            static auto dev_num = []{
                auto platform = sycl::platform{typename TPltf::selector{}};
                auto devices = platform.get_devices();
                return devices.size();
            }();
```
Otherwise you can run into troubles if two threads execute this function for the first time in parallel.